sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -69,6 +69,10 @@ case class DataSource(
 
   lazy val providingClass: Class[_] = lookupDataSource(className)
 
+  lazy private val _schemaAndName = sourceSchema()
```

Comment:
nit: this could be just one line: `lazy val (sourceName, schema) = sourceSchema()`